machines/store.ts

```diff
@@ -325,15 +354,24 @@ export async function getItem(
   encryptionKey: string
 ) {
   try {
-    const data = await Storage.getItem(key);
+    const data = await Storage.getItem(key, encryptionKey);
     if (data != null) {
       const decryptedData = decryptJson(encryptionKey, data);
       return JSON.parse(decryptedData);
+    }
+    if (data === null && vcKeyRegExp.exec(key)) {
+      await Storage.removeItem(key);
```

Comment:
Considering the "Single Responsibiliy", the getItem method is supposed have implementation only to get the elements. can we remove the removeItem Implementation from here and call store.removeItem() method here, where the remove Item functionality can be handled.

Reply:
Since store.removeItem() expects data also as a parameter to be passed. Since in this case data is null calling that method will throw error. 
One way around is adding a method for that in store.ts.

 Should i add a method for that in store.ts?

Reply:
If this is breaking SRP, shouldn't we be like throw some error or so stating about the VC tamper & then it is the responsibility of the caller to remove the respective VC?

Reply:
Summarising the discussion with @vijay151096  here,
The method getItem here is performing 2 responsibilities
1. get the item from storage
2. Remove VC in case if it is tampered
As per the SRP, we should be handling only one responsibility here. 

So , this had to be like just get the item from storage & in case if VC is tampered getItem throws an error stating "VC tamper". Then it becomes the responsibility of the caller to perform afterEffects of VC tamper i.e., delete VC data & its meta data.